libs/utils/utils.js

```diff
@@ -724,10 +724,25 @@ async function decorateIcons(area, config) {
 async function decoratePlaceholders(area, config) {
   const el = area.querySelector('main') || area;
   const regex = /{{(.*?)}}|%7B%7B(.*?)%7D%7D/g;
-  const found = regex.test(el.innerHTML);
-  if (!found) return;
+  const walker = document.createTreeWalker(
+    el,
+    NodeFilter.SHOW_TEXT,
+    {
+      acceptNode: (node) => {
+        regex.test(node.nodeValue) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_REJECT; 
+      },
+    }
+  );
+  const nodes = [];
+  let node;
+  while (node = walker.nextNode()) {
```

Comment:
I'm a bit confused - why are we using `nextNode` and not `currentNode` for the actual node we're interested in?